filesystem/storage/DBFile.php

```diff
@@ -66,36 +73,34 @@ public function forTemplate() {
 	 * @return string
 	 */
 	public function getTag() {
-		// Check mime type
-		$mime = $this->getMimeType();
-		if(empty($mime)) {
-			return '';
-		}
-
-		// Check that path is available
-		$url = $this->getURL();
-		if(empty($url)) {
-			return '';
-		}
-
-		$template = $this->getTemplateForMime($mime);
+		$template = $this->getFrontendTemplate();
 		if(empty($template)) {
 			return '';
 		}
-
-		// Render
 		return (string)$this->renderWith($template);
 	}
 
 	/**
-	 * Given a mime type, determine the template to render as on the frontend
+	 * Determine the template to render as on the frontend
 	 *
-	 * @param string $mimetype
 	 * @return string Name of template
 	 */
-	protected function getTemplateForMime($mimetype) {
+	public function getFrontendTemplate() {
```

Comment:
Because since it acts on the current object its a "get template for this" not "get template for the given mime".